mdp/nodes/gsfa_nodes.py

```diff
@@ -0,0 +1,2040 @@
+######################################################################################################################
+# gsfa_nodes: This module implements the Graph-Based SFA Node (GSFANode) and the Information-Preserving GSFA Node    #
+#             (iGSFANode).                                                                                           #
+#                                                                                                                    #
+# See the following publications for details on GSFA and iGSFA:                                                      #
+# * Escalante-B A.-N., Wiskott L, "How to solve classification and regression problems on high-dimensional data with #
+# a supervised extension of Slow Feature Analysis". Journal of Machine Learning Research 14:3683-3719, 2013          #
+# * Escalante-B., A.-N. and Wiskott, L., "Improved graph-based {SFA}: Information preservation complements the       #
+# slowness principle", e-print arXiv:1601.03945, http://arxiv.org/abs/1601.03945, 2017                               #
+#                                                                                                                    #
+# Examples of using GSFA and iGSFA are provided at the end of the file                                               #
+######################################################################################################################
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+
+import scipy
+import scipy.optimize
+import sys
+
+# NodeException, TrainingException, ValueError
+import mdp
+from mdp import numx, NodeException, TrainingException
+# from mdp.nodes.sfa_nodes import SFANode
```

Comment:
I'm not sure whether it is official MDP styleguide, but usually out-commented code should be avoided.
I'm personally fine with out-commented code if it serves some documentary purpose, e.g. document good-looking yet subtly wrong approach, too naive approach, a pitfall, something a future maintainer might be intrigued to break by an attempt to "simplify".
In that case it should however be accompanied by an explaining comment.

There are some more out-commented code snippets below. Please check those accordingly.